src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1123,6 +1123,10 @@ internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration
                     scriptConfig.Functions.Add((string)function);
                 }
             }
+            else
+            {
+                scriptConfig.Functions = null;
```

Comment:
Today if I have 50 functions and I add `"functions": ["SomeFunction"]` to my host.json -- it only runs that test.
But then if I delete that property completely, which causes a host restart -- it still only runs that test when the host restarts. That's b/c it re-uses the same ScriptConfig instance. It sees that the host.json has no `functions` property, so it does nothing -- which leaves it set to the previous value.

I was writing a test for this Http disposal thing and found this.

This change puts it back to the 'initialized' state. I could just clear the collection, but I already know 'null' works and has no odd side behaviors since we have tests for it.